atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TableTransactionConflictManager.java

```diff
@@ -0,0 +1,260 @@
+            Set<TableReference> tablesWritten) {
+        try {
+            if (!runningTransactions.containsKey(startTimestamp)) {
+                cleanUp();
+                return Status.ERROR;
+            }
+            for (TableReference write : tablesWritten) {
+                registerWriteConflicts(startTimestamp, write);
+            }
+
+            for (TableReference read : tablesRead) {
+                if (isConflictingRead(startTimestamp, read)) {
+                    runningTransactions.remove(startTimestamp);
+                    cleanUp();
+                    return Status.CANNOT_ADVISE;
+                }
+            }
+
+            TransactionState state = runningTransactions.get(startTimestamp);
+            if (state.inConflict() && state.outConflict()) {
+                runningTransactions.remove(startTimestamp);
+                cleanUp();
+                return Status.CANNOT_ADVISE;
+            }
+
+            runningTransactions.put(startTimestamp, state.withWriteSet(tablesWritten)
+                    .withReadSet(tablesRead)
+                    .withCommitTimestamp(Optional.of(commitTimestamp)));
+            tablesWritten.forEach(write -> lastUpdated.put(write, startTimestamp));
+            tablesRead.forEach(read -> reads.put(read, startTimestamp));
+            cleanUp();
+            return Status.NO_READ_WRITE_CONFLICTS;
+        } catch (RuntimeException e) {
+            log.warn("Error thrown while processing conflicts", e);
+            return Status.ERROR;
+        }
+    }
+
+    private boolean isConflictingRead(long startTimestamp, TableReference table) {
+        TransactionState ourState = runningTransactions.get(startTimestamp);
+        Set<Long> otherWriteStartTimestamps = lastUpdated.get(table);
+        for (Long otherStartTimestamp : otherWriteStartTimestamps) {
+            if (otherStartTimestamp == startTimestamp) {
+                continue;
+            }
+            TransactionState otherState = runningTransactions.get(otherStartTimestamp);
+            if (otherState.commitTimestamp().get() > startTimestamp && otherState.outConflict()) {
+                return true;
+            } else {
+                runningTransactions.put(otherStartTimestamp, otherState.withInConflict(true));
+                ourState = ourState.withOutConflict(true);
+                runningTransactions.put(startTimestamp, ourState);
+            }
+        }
+        return false;
+    }
+
+    private void registerWriteConflicts(long startTimestamp, TableReference table) {
+        Set<Long> readStartTimestamps = reads.get(table);
+        for (Long readStartTimestamp : readStartTimestamps) {
+            if (readStartTimestamp == startTimestamp) {
+                continue;
+            }
+            TransactionState readState = runningTransactions.get(readStartTimestamp);
+            if (readState.commitTimestamp().get() > startTimestamp && readState.inConflict()) {
+                continue;
```

Comment:
Discussed offline: 
- Because of how Atlas does write-write conflict checking before this point, and registration of reads and writes happens atomically for each transaction, this case may not be expected to occur.
- Would be good to get a general picture of the numbers first. We can (and of course must) prove that this is correct before actually using this to skip conflict checks.